pycbc/filter/qtransform.py

```diff
@@ -0,0 +1,254 @@
+# under the terms of the GNU General Public License as published by the
+# Free Software Foundation; either version 3 of the License, or (at your
+# option) any later version.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General
+# Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, write to the Free Software Foundation, Inc.,
+# 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+
+
+#
+# =============================================================================
+#
+#                                   Preamble
+#
+# =============================================================================
+#
+
+"""
+This module retrives a timeseries and then calculates the q-transform of that time series
+
+Example
+-------
+    $ python q-transform.py -s 4096 -u test -o /Users/pycbc_qtransform
+
+"""
+
+from math import pi, ceil, log, exp
+import numpy as np
+from pycbc.types.timeseries import FrequencySeries, TimeSeries
+import os, sys
+from pycbc.frame import read_frame
+from pycbc.filter import highpass_fir, matched_filter
+from pycbc.waveform import get_fd_waveform
+from pycbc.psd import welch, interpolate
+from pycbc.fft import ifft
+import urllib
+import datetime
+from scipy.interpolate import (interp2d, InterpolatedUnivariateSpline)
+from numpy import fft as npfft
+import argparse
+import datetime
+
+from matplotlib import use
+use('Agg')
+from matplotlib import pyplot as plt
+from matplotlib.colors import BoundaryNorm
+from matplotlib.ticker import MaxNLocator
+from matplotlib.pyplot import specgram
+
+__author__ = 'Hunter Gabbard <hunter.gabbard@ligo.org>'
+__credits__ = 'Duncan Macleod <duncan.macleod@ligo.org>'
+
+def padding(window_size, dur, f0, Q):
+    """The `(left, right)` padding required for the IFFT
+
+    Parameters
+    ----------
+    window_size: int
+        Size of window
+    dur: int
+        Duration of timeseries in seconds
+    f0: int
+        Central frequency
+    Q: int
+        q value
+
+    Returns
+    -------
+    tuple
+       Number of values padded to the edges of each axis. 
+ 
+    """
+
+    pad = n_tiles(dur,f0,Q) - window_size
+    return (int((pad - 1)/2.), int((pad + 1)/2.))
+
+def get_data_indices(dur, f0, indices):
+    """Returns the index array of interesting frequencies for this row
+
+    Parameters
+    ----------
+    dur: int
+        Duration of timeseries in seconds
+    f0: int
+        Central frequency
+    indices: numpy.ndarray
+        window indices for fft 
+
+    Returns
+    -------
+    numpy.ndarray
+        Returns index array of interesting frequencies for this row
+
+    """
+    return np.round(indices + 1 +
+                       f0 * dur).astype(int)
+
+def _get_indices(dur):
```

Comment:
Yeah it looks like I only use it once, but I think it's a nice function that I'd like to keep around in case I want to use it more than once later on.